Tests/Find-ImageName.Tests.ps1

```diff
@@ -1,13 +1,101 @@
 Import-Module -Force $PSScriptRoot/../Docker.Build.psm1
 
+function Get-TempPath {
+    if ($IsWindows) {
+        [system.io.path]::GetTempPath()
+    } elseif ($IsLinux) {
+        '/tmp'
+    }
+}
+
+function New-RandomFolder {
+    param (
+        [int]$FolderLength = 8
+    )
+    do {
+        $randomString = ( -join ((0x30..0x39) + ( 0x41..0x5A) + ( 0x61..0x7A) | Get-Random -Count $FolderLength | % {[char]$_}) )
+        $randomPath = Join-Path $(Get-TempPath) $randomString
+    } while (Test-Path -Path $randomPath -PathType Container)
+    New-Item -Path $randomPath -ItemType Directory | Out-Null
+    return $randomPath
+}
+
+function Fake-GitRepository {
+    param (
+        [ValidateNotNullOrEmpty()]
+        [String]$Path
+    )
+
+    $dotGitPath = Join-Path $Path ".git"
+    if (Test-Path $dotGitPath -PathType Container) {
+        Remove-Item $dotGitPath -Recurse -Force | Out-Null
+    }
+    New-Item $dotGitPath -ItemType Directory
+
+    $configData = @"
+[remote "origin"]
+url = https://github.com/3shapeAS/dockerbuild-pwsh.git
+"@
+
+    $configData | Out-File -FilePath "$dotGitPath/config" -Encoding ascii
+}
+
 Describe 'Parse context from git repository' {
 
+    Context 'Validating support functions for - When git is installed' {
+
+        It 'can get TEMP path' {
+            $tempPath = Get-TempPath
+
+            if ($IsWindows) {
+                $tempPath.IndexOf($Env:TEMP) | Should -Be 0
+            } elseif ($IsLinux) {
+                $tempPath.IndexOf('/tmp') | Should -Be 0
+            }
+        }
+
+        It 'can create random folder in TEMP folder' {
+            $pathInTemp = New-RandomFolder
+
+            Test-Path -Path $pathInTemp | Should -Be $true
+
+            Remove-Item $pathInTemp -Recurse -Force | Out-Null
```

Comment:
This sort of teardown code probably belongs in an AfterAll block.

Reply:
I would love to, but i need to get access to the `$tempFolder` variable during build up and tear down.

Reply:
Got it. Worked as expected.